js/auth.jsx

```diff
@@ -36,35 +35,38 @@ var auth = {
   },
 
   getCurrentUser: function(cb) {
-    function getCurrentUserLocal(cb) {
+    // Attempt to get the current user locally and check the server if that fails
+    var currentUser = getCurrentUserLocal();
+    if (!currentUser) {
+      return getCurrentUserRemote(cb);
+    } else {
+      cb(currentUser);
+    }
+
+    function getCurrentUserLocal() {
       if (typeof window.localStorage === 'undefined') {
-        return false;
-      }
-      var currentUser = JSON.parse(window.localStorage.getItem('currentUser'));
-      if (!currentUser) {
-        return false;
-      } else {
-        cb(currentUser);
-        return true;
+        return null;
       }
+      return JSON.parse(window.localStorage.getItem('currentUser'));
     }
 
     function getCurrentUserRemote(cb) {
       $.ajax({
         url:  '/api/user',
         success: function(user) {
-          window.localStorage.setItem('currentUser', JSON.stringify(user));
-          cb(user);
+          if (user) {
+            window.localStorage.setItem('currentUser', JSON.stringify(user));
+            cb(user);
+          } else {
+            cb(null);
+          }
         },
         error: function(xhr, status, err) {
-          console.error(this.props.userId, status, err.toString());
+          console.error(xhr.responseText);
```

Comment:
There are a probably a bunch of places we should change this. I used it in one of my recent components but haven't gone back and fixed it on the others.